sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1098,7 +1098,15 @@ def migrate(
         )
 
     def delete(self, name: str) -> None:
-        self.adapter.drop_view(name)
+        try:
+            self.adapter.drop_view(name)
+        except Exception:
+            logger.debug(
+                "Failed to drop view '%s'. Trying to drop the materialized view instead",
+                name,
+                exc_info=True,
+            )
+            self.adapter.drop_view(name, materialized=True)
```

Comment:
try catch this too?? or no

Reply:
If we're cleaning up a view and neither call succeeded than something is definitely wrong and we should let the error bubble up.